fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,227 @@
+            };
+
+            // Register this NATS implementation as a "subscriber" with our Notification Service.
+            // This means that our "notify" method will be called when the server publishes an event.
+            service.subscribe(this);
+            log.info("Initialized NATS publisher for channel '" + channelName + "' using servers: '" + servers + "'.");
+        } catch (Throwable t) {
+            String msg = "Caught exception while initializing NATS publisher.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "init");
+        }
+    }
+
+    /**
+     * Performs any necessary "shutdown" logic to disconnect from the channel.
+     */
+    public void shutdown() {
+        log.entering(this.getClass().getName(), "shutdown");
+
+        try {
+            if (log.isLoggable(Level.FINE)) {   
+                log.fine("Shutting down NATS publisher for channel: '" + channelName + "'.");
+            }
+            if (sc != null) {
+               sc.close();
+            }
+        } catch (Throwable t) {
+            String msg = "Caught exception shutting down NATS publisher for channel: '" + channelName + "'.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "shutdown");
+        }
+    }
+
+    /**
+     * Publishes an event to NATS.
+     */
+    @Override
+    public void notify(FHIRNotificationEvent event) throws FHIRNotificationException {
+        log.entering(this.getClass().getName(), "notify");
+        String jsonString = null;
+        try {
+            jsonString = FHIRNotificationUtil.toJsonString(event, true);
+
+            if (log.isLoggable(Level.FINE)) { 
+                log.fine("Publishing NATS notification event to channel '" + channelName + "',\nmessage: '" + jsonString + "'.");
+            }
+            
+            sc.publish("FHIRNotificationEvent", jsonString.getBytes(), acb);
+    
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Published NATS notification event to channel '" + channelName + "'");
+            }
+        } catch (Throwable e) {
+            String msg = buildNotificationErrorMessage(channelName, (jsonString == null ? "<null>" : jsonString));
+            log.log(Level.SEVERE, msg , e);
+            throw new FHIRNotificationException(msg, e);
+        } finally {
+            log.exiting(this.getClass().getName(), "notify");
+        }
+    }
+    
+    /**
+     * Builds a formatted error message to indicate a notification publication failure.
+     */
+    private String buildNotificationErrorMessage(String channelName, String notificationEvent) {
+        return String.format("NATS publication failure; channel '%s'\nNotification event: '%s'\n.", channelName, notificationEvent);
+    }
+}
```

Comment:
```suggestion
    /*
     * Modified from original NATS documentation @ https://docs.nats.io/developing-with-nats/security/tls
     * openssl is used to generate a pkcs12 file (.p12) from client-cert.pem and client-key.pem.
     * The resulting file is then imported into a java keystore using keytool which is part of java jdk.
     * keytool is also used to import the CA certificate rootCA.pem into the truststore.
     */
    private static KeyStore loadKeystore(String path, String password) throws Exception {
        KeyStore store = KeyStore.getInstance("PKCS12");
        try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(path));) {
            store.load(in, password.toCharArray());
        }
        return store;
    }

    private static KeyManager[] createKeyManagers(Properties tlsProps) throws Exception {
        KeyStore store = loadKeystore(tlsProps.getProperty("keystore"), tlsProps.getProperty("keystore-pw"));
        KeyManagerFactory factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
        factory.init(store, tlsProps.getProperty("keystore-pw").toCharArray());
        return factory.getKeyManagers();
    }

    private static TrustManager[] createTrustManagers(Properties tlsProps) throws Exception {
        KeyStore store = loadKeystore(tlsProps.getProperty("truststore"), tlsProps.getProperty("truststore-pw"));
        TrustManagerFactory factory = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
        factory.init(store);
        return factory.getTrustManagers();
    }

    private static SSLContext createSSLContext(Properties tlsProps) throws Exception {
        SSLContext ctx = SSLContext.getInstance("TLSv1.2");
        ctx.init(createKeyManagers(tlsProps), createTrustManagers(tlsProps), new SecureRandom());
        return ctx;
    }
}
```

Reply:
I tried doing a wider edit, but ran into github comments issues.  apologies for deleting another one. 
This should address my concerns.